File: local_examples/cmds_string/node-redis/cmds-string.js
// EXAMPLE: cmds_string
// HIDE_START
import assert from 'node:assert';
import { createClient } from 'redis';

const client = createClient();
await client.connect().catch(console.error);
// HIDE_END

// REMOVE_START
await client.del(['key1', 'key2', 'mykey', 'nonexisting']);
// REMOVE_END

// STEP_START mget
await client.set('key1', 'Hello');
await client.set('key2', 'World');

const mgetResult = await client.mGet(['key1', 'key2', 'nonexisting']);
console.log(mgetResult); // >>> [ 'Hello', 'World', null ]
// STEP_END

// REMOVE_START
assert.deepEqual(mgetResult, ['Hello', 'World', null]);
await client.del(['key1', 'key2', 'nonexisting']);
// REMOVE_END

// STEP_START incr
const incrResult1 = await client.set('mykey', '10');
console.log(incrResult1); // >>> OK

const incrResult2 = await client.incr('mykey');
console.log(incrResult2); // >>> 11

const incrResult3 = await client.get('mykey');
console.log(incrResult3); // >>> 11
// STEP_END

// REMOVE_START
assert.equal(incrResult1, 'OK');
assert.equal(incrResult2, 11);
assert.equal(incrResult3, '11');
await client.del('mykey');
// REMOVE_END

// HIDE_START
await client.close();
// HIDE_END

File: local_examples/cmds_string/rust-async/cmds_string.rs
// EXAMPLE: cmds_string
#[cfg(test)]
mod cmds_string_tests {
    use redis::AsyncCommands;

    #[tokio::test]
    async fn run() {
        let mut r = match redis::Client::open("redis://127.0.0.1") {
            Ok(client) => match client.get_multiplexed_async_connection().await {
                Ok(conn) => conn,
                Err(e) => {
                    println!("Failed to connect to Redis: {e}");
                    return;
                }
            },
            Err(e) => {
                println!("Failed to create Redis client: {e}");
                return;
            }
        };

        // REMOVE_START
        let _: Result<i32, _> = r.del(&["key1", "key2", "mykey", "nonexisting"]).await;
        // REMOVE_END

        // STEP_START mget
        if let Ok(res) = r.set("key1", "Hello").await {
            let _: String = res;
        }

        if let Ok(res) = r.set("key2", "World").await {
            let _: String = res;
        }

        match r.mget(&["key1", "key2", "nonexisting"]).await {
            Ok(mget_result) => {
                let mget_result: Vec<Option<String>> = mget_result;
                println!("{mget_result:?}");    // >>> [Some("Hello"), Some("World"), None]
                // REMOVE_START
                assert_eq!(
                    mget_result,
                    vec![Some("Hello".to_string()), Some("World".to_string()), None]
                );
                // REMOVE_END
            }
            Err(e) => {
                println!("Error getting values: {e}");
            }
        }
        // STEP_END

        // STEP_START incr
        if let Ok(res) = r.set("mykey", "10").await {
            let res: String = res;
            println!("{res}");    // >>> OK
            // REMOVE_START
            assert_eq!(res, "OK");
            // REMOVE_END
        }

        match r.incr("mykey", 1).await {
            Ok(incr_result) => {
                let incr_result: i64 = incr_result;
                println!("{incr_result}");    // >>> 11
                // REMOVE_START
                assert_eq!(incr_result, 11);
                // REMOVE_END
            }
            Err(e) => {
                println!("Error incrementing value: {e}");
            }
        }

        match r.get("mykey").await {
            Ok(get_result) => {
                let get_result: String = get_result;
                println!("{get_result}");    // >>> 11
                // REMOVE_START
                assert_eq!(get_result, "11");
                // REMOVE_END
            }
            Err(e) => {
                println!("Error getting value: {e}");
            }
        }
        // STEP_END

        // REMOVE_START
        let _: Result<i32, _> = r.del(&["key1", "key2", "mykey", "nonexisting"]).await;
        // REMOVE_END
    }
}
